core/client/db/src/lib.rs

```diff
@@ -408,35 +412,29 @@ impl<Block: BlockT> Backend<Block> {
 			storage: Arc::new(storage_db),
 			tries_change_storage: tries_change_storage,
 			blockchain,
-			pruning_window,
+			canonicalization_delay,
 		})
 	}
 
-	// write stuff to a transaction after a new block is finalized.
-	// this manages state pruning. Fails if called with a block which
-	// was not a child of the last finalized block.
-	fn note_finalized(
+	// performs state pruning after importning a non-finalized block.
+	fn state_pruning_non_finalized(
```

Comment:
I'd call it `force_canonicalize` or something like that. Pruning follows canonicalisation, but technically is independent. I.e. `hash` is canonicalized here, but some other older block (or a few blocks) is pruned.